src/Models/ModelInstantiator.php

```diff
@@ -224,6 +225,22 @@ public function databaseModelToDataObject(
             // Many-to-one relationship (a single nested Data Object).
             else
             {
+                // If the value is null make sure it's allowed to be null.
+                if (is_null($model_relations[$alias]))
+                {
+                    $reflection_index    = array_search($property, array_column($property_reflection, 'name'));
+                    $reflection_property = $property_reflection[$reflection_index];
+
+                    if ($reflection_property->getType()->allowsNull())
+                    {
+                        $data_object->{$property} = null;
+                        continue;
+                    }
+
+                    // Null not allowed, throw exception.
+                    throw new InvalidRelationshipDataException("{$alias} relationship must not be null.");
```

Comment:
Adding this to the `@throws` declaration in the method PHP Doc would help communicate that this can happen.

Reply:
Good call, thanks Tim!